src/ops/mod.rs

```diff
@@ -0,0 +1,4 @@
+mod array;
+mod factory;
+
+pub use array::*;
```

Comment:
I think this line is not needed. The `array` mod and `factory` mod simply provide impls, and all those pub associated functions are already exposed to the user